server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -255,7 +256,8 @@
      * Logger initialized in the ctor because if it were initialized statically
      * then it wouldn't get the node name.
      */
-    private final Logger logger;
+    private final Logger logger = LogManager.getLogger(Node.class);
```

Comment:
Doesn't this conflict with the comment just above?

Reply:
Good catch, I think the comment is unclear (since it refers to *static* initialization, which is not what is happening here, this is merely instance field initialization, a normal part of instance construction. That is, I read the comment thinking it was explaining why the field is not a *static* field, as is often the case with loggers. I also think this comment should be located where the initialization *needs* to take place, rather than on the field itself (since that's where someone is going to see the initialization and wonder why?).

Reply:
Actually, I still think the change is correct and it's the comment that is unclear and confusing. I think the key is in fact that it be an *instance* field and not *static*. There is absolutely no practical difference between this field being initialized separately, and as the first line of the constructor.

Reply:
I'm leaving the change as-is and will clean up the comment.

Reply:
In fact, we could make this logger instance a static instance if we moved the node.name setting out of the Node class and into its own class. The current problem is that touching this setting initializes the class. If the logger were static, then it would initialize the logger too, prematurely. If the setting were moved out of this class, we could touch the setting without initializing the logger here. That should be a follow-up.

Reply:
I pushed 1053cfbcc0db18f14cb04dcb4f0544a36f2533d8.